### DISCUSSION TITLE
[OpenIdDictModule] my refresh token does not return my updated roles!

### DISCUSSION THREAD

**Discussion by ParsaGachkar** [abpframework/abp#15748] [Q&A]
anyone can guide me? i'm using OpenIdDictModule my refresh token does not return my updated roles! Is this expected behavior? 
I'm talking about roles/role claim in access token retrieved from a refresh token request. 
Heres what i expected
1- I create a user with admin role
2- get access and refresh token with any grant and offline_access scope
3- remove admin role from user in database
4- when I'm renewing my access token with refresh token, i expect the role/roles claim to be updated and return my users new roles!
if this is not the expected behavior can anyone point me the correct directions to handle this?

**Comment by maliming** [comment]
hi

> 3- remove admin role from user in database

We have a cache design, Please try to remove it from UI.
